front/app/containers/EventsPage/EventsViewer/TopBar.tsx

```diff
@@ -50,25 +51,28 @@ const FilterIcon = styled(Icon)`
 
 interface Props {
   title: string;
+  setProjectIds: (value: any) => void;
 }
 
-const TopBar = memo<Props & InjectedIntlProps>(({ title, intl }) => {
-  const theme: any = useTheme();
+const TopBar = memo<Props & InjectedIntlProps>(
+  ({ title, setProjectIds, intl }) => {
+    const theme: any = useTheme();
 
-  return (
-    <Container>
-      <Title>{title}</Title>
+    return (
+      <Container>
+        <Title>{title}</Title>
 
-      <ProjectFilterDropdownPositioner>
-        <FilterIcon name="filter-funnel" />
-        <ProjectFilterDropdown
-          title={intl.formatMessage(messages.filterDropdownTitle)}
-          onChange={console.log}
-          textColor={theme.colorText}
-        />
-      </ProjectFilterDropdownPositioner>
-    </Container>
-  );
-});
+        <ProjectFilterDropdownPositioner>
+          <FilterIcon name="filter-funnel" />
+          <ProjectFilterDropdown
+            title={intl.formatMessage(messages.filterDropdownTitle)}
+            onChange={setProjectIds}
+            textColor={theme.colorText}
+          />
+        </ProjectFilterDropdownPositioner>
+      </Container>
+    );
+  }
+);
 
 export default injectIntl(TopBar);
```

Comment:
The `ProjectFilterDropdown` is just a wrapper around the `FilterSelector` component which is used all over the application, so this would make more sense in a seperate PR. Added a Jira issue for this.

Reply:
👌